NGCHM/WebContent/javascript/DetailHeatMapDisplay.js

```diff
@@ -2859,16 +2857,32 @@ NgChm.DET.getSamplingRatio = function (axis) {
  * Search Functions Section
  ***********************************************************/
 
+NgChm.DET.cleanseSearchString = function (searchStr) {
+	var srchItems = searchStr.split(/[|;, ]+/);
+	var cleanStr = "";
+	for (var j = 0; j < srchItems.length; j++) {
+		var item = srchItems[j];
+		item = item.replace(/^\|+|\.+|\,+|\;+|\,+|\;+|\|+|\.+$/g, '');
+		if (item !== '') {
+			cleanStr = cleanStr + item + ',';
+		}
+	}
+	return cleanStr.substring(0,cleanStr.length - 1);
+}

```

Comment:
This is technically an assignment statement, and I've been bitten in the past by code following this (sometimes by a fair bit) that the compiler mistakes for a continuation of the statement and issues a very confusing error message.  So I find it safer to terminate the statement with a semicolon.

Also, if there's an issue in one of the called functions, the debugger will just list this as an anonymous function.  If it's given a name, that name will show up in the debugger call trace making it a lot easier to see at a glance how it ended up in that state.